shared/src/main/scala/coop/rchain/store/LmdbKeyValueStore.scala

```diff
@@ -57,17 +56,30 @@ final case class LmdbKeyValueStore[F[_]: Sync](
   def withWriteTxn[T](f: (Txn[ByteBuffer], Dbi[ByteBuffer]) => T): F[T] =
     withTxnSingleThread(isWrite = true)(f)
 
+  // lmdb can only work with direct buffer
+  @SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements"))
+  private def forceDirectBuffer(b: ByteBuffer): ByteBuffer =
+    if (b.isDirect) b
+    else {
+      val buffer: ByteBuffer = ByteBuffer.allocateDirect(b.remaining())
+      buffer.put(b)
+      buffer.flip()
+      buffer
```

Comment:
This can be simplified without warning suppression.
`else ByteBuffer.allocateDirect(b.remaining).put(b).flip`

This silent conversion on the critical path does not feel nice. This is LMDB implementation and it can only work with direct buffer so it makes sense to break for heap buffers.
Do we need this for normal RNode operation or just for tests?